### PR TITLE
Update gradle and kotlin

### PR DESCRIPTION
**Pull request by dephillipsmichael**
Heart snapshot is mostly a duplicate of the stair step task, so Dan and Larsson are on board with replacing it entirely with the newly branded "Heart Snapshot" task.   This PR handles both that and updating the CRF module to Android Studio 4.1 and AndroidX.

### CODE REVIEW

**Comment by dephillipsmichael** [comment]
@liujoshua should be ready for review now.  I'd suggest just pulling it down and seeing if it runs in your Android Studio, otherwise it would take too long to check all the changes in this PR. 

From a high level, I removed the git sub-module dependency for backbone, and replaced it with including reserach-stack artifact 0.5.0 in the CRF module.  I also ran Refactor->Migrate to AndroidX in Android Studio and that took care of that automatically which was awesome.

**Comment by dephillipsmichael** [comment]
The travis build is failing I think because we need the server to download /usr/local/android-sdk/build-tools/29.0.3/ and also gradle 6.5.

**Comment by nategbrown9** [comment]
Looks good. Did you add heart_snapshot.json instead of using stair_step.json so that you could specify a different schema?

**Comment by dephillipsmichael** [comment]
> Looks good. Did you add heart_snapshot.json instead of using stair_step.json so that you could specify a different schema?

@nategbrown9 that was the initial driving factor, as well as copy edits that will come about during testing, and there is also the open question on if we are going to include the Sex/DOB questions within the task itself since that is asked in MP2 at other places in the app.

File: crf/build.gradle [hunk 1]
```diff
@@ -27,17 +27,14 @@ android {
...
 dependencies {
     implementation fileTree(include: ['*.jar'], dir: 'libs')
-    
-    implementation project(':backbone'), {
-        exclude group: 'joda-time', module: 'joda-time'
-        exclude group: 'net.zetetic', module: 'android-database-sqlcipher'
-        exclude group: 'co.touchlab.squeaky'
-    }
+
+    // Dagger needs access to some RS classes
```
Comment (nategbrown9) [inline crf/build.gradle:36]: Comment is confusing since the CRF module doesn't use Dagger.

Comment (dephillipsmichael) [reply crf/build.gradle:36]: I'm confused by it too and I wrote it.  Removed.

**Comment by dephillipsmichael** [comment]
@nategbrown9 Also, the CI is failing, what I'm getting from [the error](https://travis-ci.com/github/Sage-Bionetworks/CardiorespiratoryFitness-Android/builds/224907402) is that we need to update the build server to have build tools 29.0.3, but I'm not sure I have permissions to install that.

**Comment by nategbrown9** [comment]
I'm not sure what it takes to update Travis. Hoping @liujoshua is more familiar with it.

**Review by nategbrown9 — APPROVED** [review]
Code changes look good. We just need to figure out the build issue.

File: app/src/main/java/org/sagebionetworks/research/crf/CrfSample.java [hunk 2]
```diff
@@ -74,12 +72,31 @@ private void addTask(final ViewGroup taskContainer, final Intent taskIntent,
...
+        switch (taskTitle) {
+            case trainingTaskTitle:
+                taskIntent =  CrfTaskIntentFactory.getHeartRateTrainingTaskIntent(this);
+                break;
+            case restingHrTaskTitle:
+                taskIntent = CrfTaskIntentFactory.getHeartRateMeasurementTaskIntent(this);
+                break;
+            case snapshotTaskTitle:
+                if (gender != null && birthYear != null) {
+                    taskIntent = CrfTaskIntentFactory
```
Comment (dephillipsmichael) [inline app/src/main/java/org/sagebionetworks/research/crf/CrfSample.java:92]: @nategbrown9 I added an additional function to the intent factory that lets you pass in the previously answered questions to gender and birthYear and those questions get skipped the next time you do them. This was updated to match how the flow already was on iOS.

File: crf/src/main/assets/json/heart_snapshot.json [hunk 3]
```diff
@@ -38,10 +38,10 @@
     {
       "identifier": "sexForm",
       "type": "crf_form",
-      "title": "What is your sex?",
-      "learnMoreText"    : "Why are we asking about this?",
+      "title": "What is your gender?",
```
Comment (dephillipsmichael) [inline crf/src/main/assets/json/heart_snapshot.json:41]: @nategbrown9 also fyi there was a push on iOS to change all instances of "sex" to "gender" so I went and matched all the copy that iOS has for this. I kept the back-end identifier as "sex"; however, as changing that was out of scope and iOS still has "sex" as the identifier as well.

File: crf/src/main/java/org/sagebase/crf/step/CrfStartTaskStepLayout.java [hunk 4]
```diff
@@ -180,4 +199,43 @@ public void onActivityFinished(int requestCode, int resultCode, Intent data) {
 
         }
     }
+
+    @Override
+    public void goForwardClicked(View v) {
```
Comment (dephillipsmichael) [inline crf/src/main/java/org/sagebase/crf/step/CrfStartTaskStepLayout.java:204]: @nategbrown9 last thing, I added the camera permission request when you tap "next" on the first screen of each CRF task to match iOS flow and copy.

File: crf/src/main/assets/json/heart_snapshot.json (hunk 3 above)
Comment (liujoshua) [reply crf/src/main/assets/json/heart_snapshot.json:41]: Interesting. I thought the HR recovery formulas we use use sex at birth (the studies were done decades ago)

**Review by liujoshua — APPROVED** [review]
LGTM. Thanks for bringing it up to date

**Review by liujoshua — CHANGES_REQUESTED** [review]
@dephillipsmichael actually, let me take a look at travis update

File: crf/src/main/assets/json/heart_snapshot.json (hunk 3 above)
Comment (dephillipsmichael) [reply crf/src/main/assets/json/heart_snapshot.json:41]: It's a good thought, I'm going to have Dan write up a few sentences on motive/why it's OK to make the change from "sex" to "gender" scientifically to the Jira issue, just so we have it documented.

Comment (Erin-Mounts) [reply crf/src/main/assets/json/heart_snapshot.json:41]: Studies performed decades ago would not have asked for a participant's "sex at birth" and the way they likely determined sex (probably a line on a questionnaire "Sex (m/f):") wouldn't even make a distinction between sex/sex assigned at birth/gender so it's moot anyway--they wouldn't have had the data to determine which is actually most relevant to the calculation. But documentation is good.

**Comment by dephillipsmichael** [comment]
> @dephillipsmichael actually, let me take a look at travis update

Thank you, we will likely have to make the same travis fix in MP2 as well.

File: crf/src/main/assets/json/heart_snapshot.json (hunk 3 above)
Comment (liujoshua) [reply crf/src/main/assets/json/heart_snapshot.json:41]: Good to know. I had thought they used what is on your birth certificate back then. Hopefully we'll start seeing **many** more studies to establish baseline data for all sorts of common biomarkers for populations that historically weren't adequately represented

Comment (liujoshua) [reply crf/src/main/assets/json/heart_snapshot.json:41]: Promising implications for improvement in outcomes. I've followed the BMI and waist measurement renorm studies for API and the downstream impacts on cardiovascular and diabetic risk assessment are significant.

**Comment by dephillipsmichael** [comment]
@liujoshua not sure what's going on with travis-ci, the build succeeded, but merging is blocked by "waiting for status to be reported"?

**Pull request merged**